packages/components/app/styles/components/modal.scss

```diff
@@ -0,0 +1,156 @@
+//
+// MODAL
+//
+
+@use "../mixins/focus-ring" as *;
+
+.hds-modal {
+  flex-direction: column;
+  padding: 0;
+  background: var(--token-color-surface-primary);
+  border: none;
+  border-radius: 10px;
+  box-shadow: var(--token-surface-overlay-box-shadow);
+
+  &[open] {
+    position: fixed;
+    display: flex;
+  }
+
+  &::backdrop {
+    display: none;
+  }
+}
+
+.hds-modal__overlay {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  background: var(--token-color-palette-neutral-700);
+  opacity: 0.5;
+}
```

Comment:
Good question. Because the scrappy website didn't have any elements with `z-index` set on them, the `auto` did the trick for these examples. However, this doesn't work in any project, so I did a bit of research the other day to see what's out there.

#### `design-system`
 - [`breadcrumbs` is `300`](https://github.com/hashicorp/design-system/blob/main/packages/components/app/styles/components/breadcrumb.scss#L182) (for the truncated content)
 - [`dropdown` is `2`](https://github.com/hashicorp/design-system/blob/main/packages/components/app/styles/components/dropdown.scss#L134) (with [a note](https://github.com/hashicorp/design-system/issues/114) to review these values)
 - [`tabs` is `10`](https://github.com/hashicorp/design-system/blob/main/packages/components/app/styles/components/tabs.scss#L100)

#### `cloud-ui`
 - [the navigation container is `10`](https://github.com/hashicorp/cloud-ui/blob/master/addons/hc-nav/addon/styles/components/hc-app-frame.scss#L37)
 - [the close navigation button is `15`](https://github.com/hashicorp/cloud-ui/blob/f8a3e3a9387a967654742fde2512cae816a1b79d/addons/hc-nav/addon/styles/components/container/index.scss#L101)
 - [the modal container/slot is `100`](https://github.com/hashicorp/cloud-ui/blob/master/addons/hc-nav/addon/styles/components/hc-app-frame.scss#L74)
 - [the modal component is `10`](https://github.com/hashicorp/cloud-ui/blob/b7e088b2034ba35e6ac5643964944d58e09b64a7/packages/cloud-ui-core/addon/styles/components/modal-dialog/index.scss)
 - [the footer is `5`](https://github.com/hashicorp/cloud-ui/blob/master/addons/hc-nav/addon/styles/components/hc-app-frame.scss#L97)
 - [tooltip overlays](https://github.com/hashicorp/cloud-ui/blob/9d461657715b362f42902d252ce0a0e28aad9c17/engines/v-insights/addon/styles/components/tooltip-overlay.scss) and [notifications](https://github.com/hashicorp/cloud-ui/blob/0b239795535d62acdb40886d7ad5351af3241abf/packages/cloud-ui-core/addon/styles/components/notifications/index.scss) are `100`
 - [the application loading is `1000`](https://github.com/hashicorp/cloud-ui/blob/b7e088b2034ba35e6ac5643964944d58e09b64a7/hcp/app/styles/components/application-loading/index.scss)
 - [the app frame](https://github.com/hashicorp/cloud-ui/blob/a88d9eac87c55639ec68f7a9e42f32ee0744b080/packages/cloud-ui-core/addon/styles/components/hcp-app-frame/_global-header-overrides.scss#L7) and [the code editor](https://github.com/hashicorp/cloud-ui/blob/5f8209b7ca93c62ec4da48ecf289f8b025e61b71/packages/cloud-ui-core/addon/styles/components/code-editor/index.scss#L39) are `5`

#### `structure`
- [`dropdown` is `300`](https://github.com/hashicorp/structure/blob/main/packages/pds-ember/app/styles/pds/components/dropdown/__config.scss#L16)
 - [`modal` is `10`](https://github.com/hashicorp/structure/blob/main/packages/pds-ember/app/styles/pds/components/app/modal-area/_config.scss#L5)
 - [`tooltip` is `1`](https://github.com/hashicorp/structure/blob/main/packages/pds-ember/app/styles/pds/components/tooltip/__config.scss#L43)

I can dig into other apps, but I can't really say I see a pattern.